pkgs/by-name/mo/monophony/package.nix

```
@@ -0,0 +1,78 @@
+{ lib
+, fetchFromGitLab
+, python310Packages
+, wrapGAppsHook4
+, gst_all_1
+, gobject-introspection
+, gtk4
+, yt-dlp
+, libadwaita
+, libsoup_3
+, glib-networking
+,
+}:
+let
+  # pydbus from mpris_server is constrained to python310
+  python3Packages = python310Packages;
+in
+python3Packages.buildPythonApplication rec {
+  pname = "monophony";
+  version = "2.3.0";
+  format = "other";
+
+  sourceRoot = "source/source";
+  src = fetchFromGitLab {
+    owner = "zehkira";
+    repo = "monophony";
+    rev = "v${version}";
+    hash = "sha256-n9tioIXMjUs9AmNuVLrOHJqeGhotkFlv+bTeHlsQPaM=";
+  };
+
+  pythonPath = with python3Packages; [
+    mpris_server
+    pygobject3
+    ytmusicapi
+  ];
+
+  nativeBuildInputs =
+    [
+      gobject-introspection
+      wrapGAppsHook4
+    ]
+    ++ (with python3Packages; [ ordered-set nuitka ]);
+
+  buildInputs =
+    [
+      libadwaita
+      # needed for gstreamer https
+      libsoup_3
+      glib-networking
+    ]
+    ++ (with gst_all_1; [
+      gst-plugins-base
+      gst-plugins-good
+      gstreamer
+    ]);
+
+  makeFlags = [ "prefix=$(out)" ];
+
+  preFixup = ''
+    buildPythonPath "$pythonPath"
+    gappsWrapperArgs+=(
+      --prefix PYTHONPATH : "$program_PYTHONPATH"
+      --prefix PATH : "${lib.makeBinPath [yt-dlp]}"
+      # needed for gstreamer https
+      --prefix LD_LIBRARY_PATH : "${libsoup_3.out}/lib"
+    )
+  '';
+
+  meta = with lib; {
+    homepage = "https://gitlab.com/zehkira/monophony";
+    description = "Linux app for streaming music from YouTube";
+    longDescription = "Monophony is a free and open source Linux app for streaming music from YouTube. It has no ads and does not require an account.";
+    license = licenses.agpl3Plus;
+    mainProgram = "monophony";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ quadradical robert-manchester ];
```

```suggestion
    maintainers = with maintainers; [ quadradical ];
```

it's been fun but i think my nixpkg experience is coming to an end.

---

Okay, sure, I'll remove you.

---

Done